models/vrp.rb

```diff
@@ -188,6 +189,24 @@ def self.check_consistency(hash)
       raise OptimizerWrapper::DiscordantProblemError, 'Rests are not available with periodic heuristic.' unless hash[:vehicles].all?{ |vehicle| vehicle[:rests].to_a.empty? }
     end
 
+    def self.deduce_data_from_hash(hash)
```

Comment:
This function only generates relations from shipments but its name is very generic. If we can move it inside `expand_data` then we can name it `create_relations_of_direct_shipments`.